server/src/main/java/org/opensearch/index/store/Store.java

```diff
@@ -334,6 +336,51 @@ public MetadataSnapshot getMetadata(SegmentInfos segmentInfos) throws IOExceptio
         return new MetadataSnapshot(segmentInfos, directory, logger);
     }
 
+    /**
```

Comment:
Nit - could add in a line explaining why we're leaving out the segments_n files

Reply:
Thanks @Poojita-Raj for the comment. This change is needed to fix the file not found exception.
PR against main #4366 contains more details around the issue and fix.